src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoActions } from '../../features/currentTodo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const [choosedTodoUser, setChoosedTodoUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo) || null;
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId).then(user => {
+        setChoosedTodoUser(user);
+      });
```

Comment:
Ensure that the `getUser` function handles errors gracefully. If the API call fails, consider setting a default state or displaying an error message to the user.